### PR TITLE
Update otel to v0.12.0

### PR DESCRIPTION
**Pull request by yas-okadatech**

### CODE REVIEW

**Comment by bogdandrutu** [comment]
A merge and release of this PR would be really appreciated by the opentelemetry-collector-contrib repo

**Comment by mrothroc** [comment]
It would be great to have this.  I'll just note that you can't use go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc
to trace gRPC calls in google cloud without this PR because that module requires 0.12.0

File: exporter/trace/cloudtrace.go [hunk 1]
```diff
@@ -233,26 +233,23 @@ type Exporter struct {
...
 	exporter, err := NewExporter(opts...)
 	if err != nil {
 		return nil, nil, err
 	}
-	tp, err := sdktrace.NewProvider(append(topts, sdktrace.WithSyncer(exporter))...)
-	if err != nil {
-		return nil, nil, err
-	}
+	tp := sdktrace.NewTracerProvider(append(topts, sdktrace.WithSyncer(exporter))...)
 	return tp, exporter.traceExporter.Flush, nil
```
Comment (james-bebbington) [inline exporter/trace/cloudtrace.go:253]: Would this be somewhat cleaner if we returned `exporter.Shutdown` here instead or `exporter.traceExporter.Flush`?

**Review by james-bebbington — APPROVED** [review]
LGTM other than one minor comment.

Since this PR is blocking people I will merge it now, but can you please check the comment below and fix in a new PR if correct?

**Pull request merged**